publish-image/action.yml

```diff
@@ -110,8 +110,13 @@ runs:
         DIGEST=${IMAGE_REPO_DIGEST#*@}
 
         # Construct the package url (purl)
-        # TODO (@Techassi): Can we use 'oci' instead of 'docker' as the type?
-        PURL="pkg:docker/$IMAGE_REPOSITORY@$DIGEST?repository_url=$REGISTRY_URI"
+        URLENCODED_DIGEST=$(echo "$DIGEST" | sed 's/:/%3A/g')
+        URLENCODED_IMAGE_REPOSITORY=$(echo "$IMAGE_REPOSITORY" | sed 's/\//%2F/g')
+        # Last item, split by /
+        IMAGE_NAME=$(echo "$IMAGE_REPOSITORY" | awk -F'/' '{print $NF}')
+        # Obtain architecture from container image
+        ARCH=$(docker inspect --format='{{index .Architecture}}' "${IMAGE_REPO_DIGEST}")
```

Comment:
suggestion: We already have access to the architecture via the `IMAGE_MANIFEST_TAG` env var. We could use

```shell
ARCH=$(echo "$IMAGE_MANIFEST_TAG" | awk -F'-' '{print $NF}')
```

This saves one call to `docker` which has to fetch the data from remote.

Reply:
Yep, true. However, this requires the image tag to contain the architecture, which it should, but it's not really guaranteed.
I could do it this way and then add an additional check to verify $ARCH is either "amd64" or "arm64" (and fail if it's not).

Reply:
Implemented it this way now.